### ISSUE TITLE
Using multiple b-taggers

### ISSUE THREAD

**Issue by ecorrigan** [UCATLAS/xAODAnaHelpers#1052]
Dear xAH experts,

I am generating ntuples for my analysis, which include b-tagging information. Since the b-tagging group does not yet recommend a single tagger, but instead has advised us to focus on three specific ones (the idea being that one of these will become the sole recommendation), it would be great to be able to give xAH a list of taggers somehow, rather than just one. 

I have looked at the code and my impression is that this functionality is not currently there. Am I wrong? Is there an easy way this could be implemented? If not, do you have any other suggestions? 

Thanks in advance!

Eric Corrigan

**Comment by kratsg** [comment]
@ecorrigan - if you're talking about using the JetSelector algorithm and specifying a tagger name, then all you need to do is create 3 different JetSelector algorithms and schedule them:

```python
for btag in ["MV2c10", "DL1rnn", "...."]:
  c.setalg("JetSelector", {"m_name": "BTagging_{0:s}".format(btag),
                           "m_taggerName": btag
                           ......
                          })
```

Let me know if this isn't clear.

**Comment by ecorrigan** [comment]
Hi! 

Thanks for the fast reply! Our ntuple generation code (ZprimeDM, in case you happen to be familiar) seems to do something like 

```
c.setalg("BJetEfficiencyCorrector", {"m_name": name1,
                           "m_taggerName": tagger1
                           ......
                          })
```

and I just now, as a first test, added the same again with name2 and tagger2, which seems to work. 
Do the decoration names depend on the tagger names? I will have to look into that, as you say. 

Thanks again

Eric

**Comment by kratsg** [comment]
@ecorrigan - for a standard analysis, most people tend to pick a tagger and then run over multiple working points. Therefore, we provide standard logic that appends the working point to `m_decor` automatically.. but this is something you can set as well with `BJetEfficiencyCorrector`: https://github.com/UCATLAS/xAODAnaHelpers/blob/master/Root/BJetEfficiencyCorrector.cxx#L141-L147

It will also print out what the decoration name is in the output, so you can verify.

**Comment by kkrizka** [comment]
The `BJetEfficiencyCorrector` code does work with multiple b-tagging algorithms. However the ntupler code needs to be updated to allow storing multiple decisions in the same tree. I've started work on making the b-tagging output more generic here: https://github.com/UCATLAS/xAODAnaHelpers/tree/r21btag

**Comment by kkrizka** [comment]
Closed by #1054 .
